pkg/apis/kubeflow.org/v1/pytorch_defaults.go

```diff
@@ -79,4 +79,9 @@ func SetDefaults_PyTorchJob(job *PyTorchJob) {
 	}
 	// Set default elastic policy.
 	setElasticPolicy(job)
+
+	if job.Spec.NprocPerNode == nil {
```

Comment:
Can we check elasticPolicy? if elasticPolicy.NProcPerNode is set, `validateNprocPerNode` rejects to create the Job, right?

Reply:
Also, can you add a unit test?

Reply:
I'm little bit confused here, for now, we are going to just leave nprocPerNode in elasticPolicy work with some warning and deprecate it in the future or it will not work since this version ?

Reply:
@kuizhiqing Sorry for the confusion.
I meant the following checks:

```suggestion
	if (job.Spec.ElasticPolicy != nil && job.Spec.ElasticPolicy.NProcPerNode == nil) || (job.Spec.ElasticPolicy == nil) {
		if job.Spec.NprocPerNode == nil {
```

If we don't check elastciPolicy and then both `elastciPolicy.NProcPerNode` and `job.Spec.NprocPerNode` are set, the `validateNprocPerNode` function rejects the request, right?

Reply:
@tenzen-y OK,Thanks for clarifying. I will handle this and those UT as you say next week.

Reply:
Done